File: Erik_Vincent/Assignment1/Assignment1/Parser/QLSParser.cs
using Antlr4.Runtime;
using Assignment1.Model.QLS.AST;

namespace Assignment1.Parser
{
    public static class QLSParser
    {
        public static StyleSheet ParseString(string input)
        {
            ICharStream stream = CharStreams.fromstring(input);
            ITokenSource lexer = new QLSLexer(stream);
            ITokenStream tokens = new CommonTokenStream(lexer);
            var parser = new QLS(tokens);
            return parser.stylesheet().result;
        }
    }
}

File: DogeQL/src/main/kotlin/qls/ast/QlsParser.kt
package qls.ast

import QuestionnaireLanguageStyleGrammarLexer
import QuestionnaireLanguageStyleGrammarParser
import org.antlr.v4.runtime.ANTLRInputStream
import org.antlr.v4.runtime.CommonTokenStream
import qls.ast.node.QlsNode
import java.io.File
import java.io.FileInputStream


class QlsParser {
    fun parse(file: File): QlsNode {
        FileInputStream(file).use {
            val stream = ANTLRInputStream(it)
            val lexer = QuestionnaireLanguageStyleGrammarLexer(stream)
            val tokens = CommonTokenStream(lexer)
            val parser = QuestionnaireLanguageStyleGrammarParser(tokens)

            val visitor = QuestionnaireLanguageStyleVisitor()

            val ast = visitor.visit(parser.stylesheet())

            return ast
        }
    }

}